nh_observations/notifications.py

```diff
@@ -97,25 +97,39 @@ class nh_clinical_notification_frequency(orm.Model):
     }
     _notifications = []
 
+    @api.constrains('observation')
+    def _check_valid_observation_model_name(self):
+        if 'nh.clinical.patient.observation' not in self.observation:
+            raise exceptions.ValidationError(
+                "Observation field assigned an invalid observation model name."
+            )
+
     def complete(self, cr, uid, activity_id, context=None):
         activity_pool = self.pool['nh.activity']
         review_frequency = activity_pool.browse(
-            cr, uid, activity_id, context=context)
+            cr, uid, activity_id, context=context
+        )
+        patient_id = review_frequency.data_ref.patient_id.id
+        observation_type = review_frequency.data_ref.observation
         domain = [
-            ('patient_id', '=', review_frequency.data_ref.patient_id.id),
-            ('data_model', '=', review_frequency.data_ref.observation),
+            ('patient_id', '=', patient_id),
```

Comment:
Might be safer to use `spell_activity_id` instead of `patient_id` as patient_id will return observations for closed spells so if a patient had a previous spell and for whatever reason a review frequency task is present this would work when it shouldn't